src/store/modules/nft.js

```diff
@@ -602,56 +562,35 @@ const actions = {
     }
     return owners;
   },
-  async fetchNFTListByAddress(
-    { commit, getters, dispatch },
-    { address, shouldFetchDetails = true } = {}
-  ) {
-    const [collectedNFTs, createdNFTClasses] = await Promise.all([
-      fetchAllNFTFromChain(this.$api, address),
-      fetchAllNFTClassFromChain(this.$api, address),
+  async fetchNFTListByAddress({ commit, dispatch }, address) {
+    const [collected, created] = await Promise.all([
+      fetchAllNFTClassFromChain(this.$api, { nftOwner: address, expand: true }),
+      fetchAllNFTClassFromChain(this.$api, { iscnOwner: address }),
     ]);
 
-    const nftClassIdDataMap = new Map();
-    collectedNFTs.forEach(n => nftClassIdDataMap.set(n.class_id, n.class_data));
-    createdNFTClasses.forEach(c => nftClassIdDataMap.set(c.id, c));
-    if (shouldFetchDetails) {
-      nftClassIdDataMap.forEach((classData, classId) => {
-        dispatch('parseAndStoreNFTClassMetadata', { classId, classData });
-      });
-    }
-
-    const formattedCreatedNFTClasses = createdNFTClasses.map(
-      formatNFTClassInfo
-    );
-    const formattedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
+    const formattedCreated = created.map(formatNFTClassInfo);
+    const formattedCollected = collected.map(formatNFTClassInfo);
     commit(TYPES.NFT_SET_USER_CLASSID_LIST_MAP, {
       address,
       nfts: {
-        created: getters.normalizeNFTList(formattedCreatedNFTClasses),
-        collected: getters.normalizeNFTList(formattedCollectedNFTs),
+        created: formattedCreated,
+        collected: formattedCollected,
       },
     });
 
-    const timestampMap = {};
-    formattedCollectedNFTs.forEach(nft => {
-      const { classId, timestamp } = nft;
-      if (!timestampMap[classId] || timestampMap[classId] < timestamp) {
-        timestampMap[classId] = timestamp;
-      }
-    });
-    commit(TYPES.NFT_SET_USER_LAST_COLLECTED_TIMESTAMP_MAP, {
-      address,
-      timestampMap,
+    const nftClassIdDataMap = new Map();
+    collected.forEach(c => nftClassIdDataMap.set(c.id, c));
+    created.forEach(c => nftClassIdDataMap.set(c.id, c));
+    nftClassIdDataMap.forEach((classData, classId) => {
+      dispatch('parseAndStoreNFTClassMetadata', { classId, classData });
```

Comment:
Should verify if the data format in class api and metadata api is matches, and should normalize them if not.

For instance, we use `iscn_owner` a lot in nft mixin, but in class API the field is returned as `owner`
same for `created_at` vs `iscn_record_timestamp`

We should store these data using consistent field names, so that external callers do not need to check if the data is fetched from class API or metadata API

Reply:
Currently, `parseAndStoreNFTClassMetadata` ignore above-mentioned fields and store common fields only.
https://github.com/likecoin/liker-land/blob/master/src/store/modules/nft.js#L497-L511
<img width="890" alt="截圖 2023-07-15 上午10 15 09" src="https://github.com/likecoin/liker-land/assets/33746295/f4065623-c930-4fad-8bd9-bc35152a56a8">

Should we further include additional fields?

Reply:
We have a computed field base on `this.NFTClassMetadata?.iscn_owner`
https://github.com/likecoin/liker-land/blob/5d9854faaeeef3a5528c5824f8195013d517bec3/src/mixins/nft.js#L178
Does not loading this field causes UI issue? Preferably we should load and normalize this field into class data via this function